### DISCUSSION TITLE
Email provider errors out.

### DISCUSSION THREAD

**Discussion by ThatOneCalculator** [nextauthjs/next-auth#3785] [Help] [answered]
See #3781 , I have no clue why my issue was closed but I guess I'll go here.

Pasted from issue:

### Description 🐜

Using the process provided in the docs (and adding `nodemailer` with `yarn`), trying to use `EmailProvider` gives an error: 
```
TypeError: Cannot destructure property 'getUserByEmail' of 'adapter' as it is undefined.
```

### Is this a bug in your own project?

Yes

### How to reproduce ☕️

1. `yarn add nodemailer`
2. Add 
```
EMAIL_SERVER=smtp://CENSORED:CENSORED@smtp.gmail.com:587
EMAIL_FROM=CENSORED@gmail.com
```
to `.env.local`
3. Add
```js
    EmailProvider({
      server: process.env.EMAIL_SERVER,
      from: process.env.EMAIL_FROM
    })
```
to `[...nextauth].js`
4. `yarn dev`
5. Attempt to fill in email

### Screenshots / Logs 📽

![ezgif-2-9da8f9ad1b](https://user-images.githubusercontent.com/44733677/152067582-c6459d0b-94b6-4310-a9b0-9b71d094602c.png)

### Environment 🖥

```
  System:
    OS: Linux 5.16 Arch
    CPU: (16) x64 AMD Ryzen 7 5800X 8-Core Processor
    Memory: 18.66 GB / 31.35 GB
    Container: Yes
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.3.0 - /usr/bin/node
    Yarn: 1.22.17 - /usr/bin/yarn
    npm: 8.4.0 - /usr/bin/npm
  Browsers:
    Firefox: 96.0.3
  npmPackages:
    next: ^12.0.7 => 12.0.9 
    next-auth: ^4.1.0 => 4.1.2 
    react: ^17.0.2 => 17.0.2 
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by jacobhq** [comment, marked as answer]
The error is very undescriptive, but the reason it shows is because you don't have a database connected to store the verification codes. I solved the error by using the prisma adaptor, and connecting a database. This is in the docs [here](https://next-auth.js.org/providers/email#:~:text=Do%20not%20forget%20to%20setup%20one%20of%20the%20database%20adapters%20for%20storing%20the%20Email%20verification%20token.).
